File: backend/WorldBeauty/Cliente.java
package WorldBeauty;

import java.util.Date;
import java.util.List;

public class Cliente {
    private String nome;
    private String nomeSocial;
    private String genero;
    private CPF CPF;
    private List<RG> RGs;
    private Date dataCadastro;
    private List<Telefone> telefones;
    private List<Produto> produtosConsumidos;
    private List<Servico> servicosConsumidos;

    Cliente(
        String nome,
        String nomeSocial,
        String genero,
        CPF CPF,
        List<RG> RGs,
        Date dataCadastro,
        List<Telefone> telefones,
        List<Produto> produtosConsumidos,
        List<Servico> servicosConsumidos
        ) {
        this.nome = nome;
        this.nomeSocial = nomeSocial;
        this.genero = genero;
        this.CPF = CPF;
        this.RGs = RGs;
        this.dataCadastro = dataCadastro;
        this.telefones = telefones;
        this.produtosConsumidos = produtosConsumidos;
        this.servicosConsumidos = servicosConsumidos;
    }

    public String GetNome() {
        return this.nome;
    }
    public String GetNomeSocial() {
        return this.nomeSocial;
    }
    public String GetGenero() {
        return this.genero.toUpperCase();
    }
    public CPF GetCPF() {
        return this.CPF;
    }
    public List<RG> GetRGs() {
        return this.RGs;
    }
    public Date GetDataCadastro() {
        return this.dataCadastro;
    }
    public List<Telefone> GetTelefones() {
        return this.telefones;
    }

    public void SetNome(String nome) {
        this.nome = nome;
    }
    public void SetNomeSocial(String nomeSocial) {
        this.nomeSocial = nomeSocial;
    }
    public void SetGenero(String genero) {
        this.genero = genero.toUpperCase();
    }
    public void SetRGs(List<RG> RGs) {
        this.RGs = RGs;
    }
    public void SetTelefones(List<Telefone> telefones) {
        this.telefones = telefones;
    }

    public List<Produto> GetProdutosConsumidos() {
        return this.produtosConsumidos;
    }
    public List<Servico> GetServicosConsumidos() {
        return this.servicosConsumidos;
    }
}

File: world-beauty/src/WB-ATV1/src/Models/Cliente.ts
import CPF from "./CPF";
import Produto from "./Produto";
import RG from "./RG";
import Servico from "./Servico";
import Telefone from "./Telefone";

export default class Cliente {
    private nome: string;
    private nomeSocial: string;
    private genero: string;
    private CPF: CPF;
    private RGs: Array<RG>;
    private dataCadastro: Date;
    private telefones: Array<Telefone>;
    private produtosConsumidos: Array<Produto>;
    private servicosConsumidos: Array<Servico>;

    constructor(
        nome: string,
        nomeSocial: string,
        genero: string,
        CPF: CPF,
        RGs: Array<RG>,
        dataCadastro: Date,
        telefones: Array<Telefone>,
        produtosConsumidos: Array<Produto>,
        servicosConsumidos: Array<Servico>
        ) {
        this.nome = nome;
        this.nomeSocial = nomeSocial;
        this.genero = genero;
        this.CPF = CPF;
        this.RGs = RGs;
        this.dataCadastro = dataCadastro;
        this.telefones = telefones;
        this.produtosConsumidos = produtosConsumidos;
        this.servicosConsumidos = servicosConsumidos;
    }

    public get GetNome(): string {
        return this.nome;
    }
    public get GetNomeSocial(): string {
        return this.nomeSocial;
    }
    public get GetGenero(): string {
        return this.genero.toUpperCase();
    }
    public get GetCPF(): CPF {
        return this.CPF;
    }
    public get GetRGs(): Array<RG> {
        return this.RGs;
    }
    public get GetDataCadastro(): Date {
        return this.dataCadastro;
    }
    public get GetTelefones(): Array<Telefone> {
        return this.telefones;
    }

    public set SetNome(nome: string) {
        this.nome = nome;
    }
    public set SetNomeSocial(nomeSocial: string) {
        this.nomeSocial = nomeSocial;
    }
    public set SetGenero(genero: string) {
        this.genero = genero.toUpperCase();
    }
    public set SetRGs(RGs: Array<RG>) {
        this.RGs = RGs;
    }
    public set SetTelefones(telefones: Array<Telefone>) {
        this.telefones = telefones;
    }

    public get GetProdutosConsumidos() {
        return this.produtosConsumidos;
    }
    public get GetServicosConsumidos() {
        return this.servicosConsumidos;
    }
}